packages/compass-sidebar/src/components/multiple-connections/saved-connections/saved-connection.tsx

```diff
@@ -0,0 +1,267 @@
+  const { connectionColorToHex } = useConnectionColor();
+
+  const isLocalhost =
+    connectionInfo.connectionOptions.connectionString.startsWith(
+      'mongodb://localhost'
+    );
+
+  const maybeProtectConnectionString = useMaybeProtectConnectionString();
+  const [hoverProps, isHovered] = useHoverState();
+  const isFavorite = connectionInfo.savedConnectionType === 'favorite';
+  const { openToast } = useToast('compass-connections');
+
+  const icon = isLocalhost ? (
+    <Icon size={spacing[3]} className={iconStyles} glyph="Laptop" />
+  ) : isFavorite ? (
+    <Icon size={spacing[3]} className={iconStyles} glyph="Favorite" />
+  ) : (
+    <ServerIcon />
+  );
+
+  async function copyConnectionString(connectionString: string) {
+    try {
+      await navigator.clipboard.writeText(connectionString);
+      openToast('copy-to-clipboard', {
+        title: 'Success',
+        description: 'Copied to clipboard.',
+        variant: 'success',
+        timeout: TOAST_TIMEOUT_MS,
+      });
+    } catch (err) {
+      openToast('copy-to-clipboard', {
+        title: 'Error',
+        description:
+          'An error occurred when copying to clipboard. Please try again.',
+        variant: 'warning',
+        timeout: TOAST_TIMEOUT_MS,
+      });
+    }
+  }
+
+  const actions: (ItemAction<Action> | ItemSeparator)[] = [
+    {
+      action: 'edit-connection',
+      label: 'Edit connection',
+      icon: 'Edit',
+    },
+    {
+      action: 'copy-connection-string',
+      label: 'Copy connection string',
+      icon: 'Copy',
+    },
+    {
+      action: 'toggle-favorite',
+      label:
+        connectionInfo.savedConnectionType === 'favorite'
+          ? 'Unfavorite'
+          : 'Favorite',
+      icon: 'Favorite',
+    },
+    {
+      action: 'duplicate-connection',
+      label: 'Duplicate',
+      icon: 'Clone',
+    },
+    { separator: true },
+    {
+      action: 'remove-connection',
+      label: 'Remove',
+      icon: 'Trash',
+      variant: 'destructive',
+    },
+  ];
+
+  const onAction = useCallback(
+    (action: Action) => {
+      switch (action) {
+        case 'edit-connection':
+          return onEditConnection(connectionInfo);
+        case 'copy-connection-string':
+          return void copyConnectionString(
+            maybeProtectConnectionString(
+              connectionInfo.connectionOptions.connectionString
+            )
+          );
+        case 'toggle-favorite':
+          return onToggleFavoriteConnection(connectionInfo);
+        case 'duplicate-connection':
+          return onDuplicateConnection(connectionInfo);
+        case 'remove-connection':
+          return onDeleteConnection(connectionInfo);
+      }
+    },
+    [
+      connectionInfo,
+      onEditConnection,
+      copyConnectionString,
+      maybeProtectConnectionString,
+      onToggleFavoriteConnection,
+      onDuplicateConnection,
+      onDeleteConnection,
+    ]
+  );
+  return (
+    <li
```

Comment:
This item is part of the compass navigation, but is completely uninteractive when using a keyboard. Are we planning to address this?

Reply:
Yes, the idea is to reuse `useVirtualNavigationTree` to implement keyboard navigation because we don't have anything specified in the design. You made me realise that I didn't create a task to apply keyboard accessibility so I'll create one for that and I'll discuss with Ben if there is any specific case we should consider on the keyboard navigation.